src/Illuminate/Database/Query/Builder.php

```diff
@@ -3915,6 +3915,26 @@ public function updateOrInsert(array $attributes, array|callable $values = [])
         return (bool) $this->limit(1)->update($values);
     }
 
+    /**
+     * Get the first record matching the attributes or insert it.
+     *
+     * @param array $attributes
+     * @param array $values
+     * @return object
+     */
+    public function firstOrInsert(array $attributes, array $values = []): object
+    {
+        $record = $this->where($attributes)->first();
+
+        if (! is_null($record)) {
```

Comment:
You can avoid the function call like this:
```suggestion
        if ($record !== null) {
```